src/NServiceBus.Core/Routing/HybridSubscriptions.cs

```diff
@@ -0,0 +1,285 @@
+﻿namespace NServiceBus.Routing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using Features;
+    using Logging;
+    using NServiceBus.Settings;
+    using Pipeline;
+    using Transport;
+    using Unicast.Messages;
+    using Unicast.Queuing;
+    using Unicast.Subscriptions.MessageDrivenSubscriptions;
+    using Unicast.Transport;
+
+    class HybridSubscriptions : Feature
+    {
+        public HybridSubscriptions()
+        {
+            EnableByDefault();
+            Prerequisite(c => c.Settings.Get<TransportInfrastructure>().OutboundRoutingPolicy.Publishes == OutboundRoutingType.Multicast, "The transport does not support native pub sub");
+            Prerequisite(c => IsMigrationModeEnabled(c.Settings), "The transport has not enabled pub sub migration mode");
```

Comment:
should that be the first prerequisite as otherwise the diagnostics would always state that Hybrid mode is deactivated because of the missing native pubsub support?

Reply:
All prerequisites are always checked, so for a unicast-only transport, you'll see both reasons.

Reply:
ah, I didn't check the behavior. That's fine then 👍